### ISSUE TITLE
bug in merge.data.table when one column is difftime

### ISSUE THREAD

**Issue by adamaltmejd** [Rdatatable/data.table#5589]
With data.table_1.14.7.

```
    dt1 <- data.table(a = as.Date(c("2022-01-01", "2022-01-02")), b = c(1,1), t = as.difftime(1, units = "days"))
    dt2 <- data.table(a = as.Date(c("2022-01-01")), c = c(1))
    dt3 <- data.table(a = as.Date(c("2022-01-01", "2022-01-03")), c = c(1,1))
```

```
> merge(dt1, dt2, all = T)
Key: <a>
            a     b          t     c
       <Date> <num> <difftime> <num>
1: 2022-01-01     1     1 days     1
2: 2022-01-02     1     1 days    NA
```

```
> merge(dt1, dt3, all = T)
Error in rbindlist(l, use.names, fill, idcol) : 
  Class attribute on column 4 of item 2 does not match with column 4 of item 1.
```

Of course, neither item 1 or item 2 have 4 columns.

**Comment by ben-schwen** [comment]
Looks like a duplicate to #5309 or #5542 and should be resolved once #5446 is merged.

This bug appeared when we changed `merge` in #5263 to use `rbind(..., use.names=FALSE, fill=TRUE)` instead of manually filling the binding objects.

**Comment by ben-schwen** [comment]
This was resolved by #5468 (checked now).
